### PR TITLE
First draft of logs interpretability

### PR DESCRIPTION
**Pull request by eladiw**

### CODE REVIEW

File: presidio-analyzer/analyzer/recognizer_result.py [hunk 1]
```diff
@@ -13,3 +14,9 @@ def __init__(self, entity_type, start, end, score):
         self.start = start
         self.end = end
         self.score = score
+        self.interpretability_details = {}
```
Comment (omri374) [inline presidio-analyzer/analyzer/recognizer_result.py]: maybe 'explanations' or 'result_description' is a better phrase than 'interpretability_details'. Interpretability probably means more to us than to the new user

File: presidio-analyzer/analyzer/nlp_engine/spacy_nlp_engine.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def __init__(self):
         self.logger = logging.getLogger(__name__)
         self.logger.setLevel(loglevel)
 
-        self.logger.info("Loading NLP model...")
+        logging.info("Loading NLP model...", extra={'req_id': ''})
```
Comment (omri374) [inline presidio-analyzer/analyzer/nlp_engine/spacy_nlp_engine.py]: can req_id have a default value? We could keep all non-request-related logs as is

File: presidio-analyzer/analyzer/analyzer_engine.py [hunk 3]
```diff
@@ -41,7 +62,13 @@ def Apply(self, request, context):
         # pylint: disable=no-member
         response.analyzeResults.extend(
             AnalyzerEngine.__convert_results_to_proto(results))
-        logging.info("Found %d results", len(results))
+
+        for res in results:
```
Comment (omri374) [inline presidio-analyzer/analyzer/analyzer_engine.py]: wrap in function

File: presidio-analyzer/analyzer/analyzer_engine.py [hunk 4]
```diff
@@ -118,6 +147,9 @@ def analyze(self, text, entities, language, all_fields):
             # analyze using the current recognizer and append the results
             current_results = recognizer.analyze(text, entities, nlp_artifacts)
             if current_results:
+                for cur_res in current_results:
```
Comment (omri374) [inline presidio-analyzer/analyzer/analyzer_engine.py]: I'm not sure I understand this part. Are we putting the recognizer name as a part of the result description? Can it be done when the RecognizerResult is created inside a recognizer?

File: presidio-analyzer/analyzer/recognizer_result.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 class RecognizerResult:
 
-    def __init__(self, entity_type, start, end, score):
+    def __init__(self, entity_type, start, end, score,
+                 interpretability_details=None):
```
Comment (omri374) [inline presidio-analyzer/analyzer/recognizer_result.py]: add docstring for interpretability_details

File: presidio-analyzer/analyzer/recognizer_result.py (hunk 1 above)
Comment (eladiw) [reply presidio-analyzer/analyzer/recognizer_result.py]: 👍

File: presidio-analyzer/analyzer/nlp_engine/spacy_nlp_engine.py (hunk 2 above)
Comment (eladiw) [reply presidio-analyzer/analyzer/nlp_engine/spacy_nlp_engine.py]: a) no it is not possible unfortunately.
b) I chose a different path so (a) is no longer an issue

**Pull request closed without merging**